src/reps/object.js

```diff
@@ -1,173 +1,174 @@
+      }, " { "),
+      ...propsArray,
+      safeObjectLink(props, {
+        className: "objectRightBrace",
+      }, " }")
+    )
+  );
+}
 
-    return propsArray;
-  },
+function getTitle(props, object) {
+  let title = props.title || object.class || "Object";
+  return safeObjectLink(props, {className: "objectTitle"}, title);
+}
 
-  getPropsArray: function (object, max, filter) {
-    let propsArray = [];
+function safePropIterator(props, object, max) {
+  max = (typeof max === "undefined") ? 3 : max;
+  try {
+    return propIterator(props, object, max);
+  } catch (err) {
+    console.error(err);
+  }
+  return [];
+}
 
-    max = max || 3;
-    if (!object) {
-      return propsArray;
-    }
+function propIterator(props, object, max) {
+  let isInterestingProp = (t, value) => {
+    // Do not pick objects, it could cause recursion.
+    return (t == "boolean" || t == "number" || (t == "string" && value));
+  };
+
+  // Work around https://bugzilla.mozilla.org/show_bug.cgi?id=945377
+  if (Object.prototype.toString.call(object) === "[object Generator]") {
+    object = Object.getPrototypeOf(object);
+  }
+
+  // Object members with non-empty values are preferred since it gives the
+  // user a better overview of the object.
+  let interestingObject = getFilteredObject(object, max, isInterestingProp);
+
+  if (Object.keys(interestingObject).length < max) {
+    // There are not enough props yet (or at least, not enough props to
+    // be able to know whether we should print "more…" or not).
+    // Let's display also empty members and functions.
+    interestingObject = Object.assign({}, interestingObject,
+      getFilteredObject(
+        object,
+        max - Object.keys(interestingObject).length,
+        (t, value) => !isInterestingProp(t, value)
+      )
+    );
+  }
+
+  const truncate = Object.keys(object).length > max;
+  let propsArray = getPropsArray(interestingObject, truncate);
+  if (truncate) {
+    propsArray.push(Caption({
+      object: safeObjectLink(props, {},
+        (Object.keys(object).length - max) + " more…")
+    }));
+  }
+
+  return propsArray;
+}
 
-    // Hardcode tiny mode to avoid recursive handling.
-    let mode = MODE.TINY;
-
-    try {
-      for (let name in object) {
-        if (propsArray.length > max) {
-          return propsArray;
-        }
-
-        let value;
-        try {
-          value = object[name];
-        } catch (exc) {
-          continue;
-        }
-
-        let t = typeof value;
-        if (filter(t, value)) {
-          propsArray.push(PropRep({
-            mode: mode,
-            name: name,
-            object: value,
-            equal: ": ",
-            delim: ", ",
-          }));
-        }
-      }
-    } catch (err) {
-      console.error(err);
-    }
+/**
+ * Get an array of components representing the properties of the object
+ *
+ * @param {Object} object
+ * @param {Boolean} truncate true if the object will be truncated.
```

Comment:
I think it's more accurate to say that truncate is true if the object has been truncated (and maybe rename to `truncated` ?). We actually truncate when building "interestingObject" and depending on that, we decide to use a comma or not for the last item.